server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -308,6 +309,87 @@ public SourceFieldMapper.Mode defaultSourceMode() {
         public String getDefaultCodec() {
             return CodecService.BEST_COMPRESSION_CODEC;
         }
+    },
+    LOOKUP("lookup") {
+        @Override
+        void validateWithOtherSettings(Map<Setting<?>, Object> settings) {
+            Integer defaultNumberOfShards = IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.get(Settings.EMPTY);
+            Object providedNumberOfShards = settings.getOrDefault(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING, defaultNumberOfShards);
+            if (Objects.equals(providedNumberOfShards, defaultNumberOfShards) == false) {
```

Comment:
Maybe just check whether `providedNumberOfShards` is 1?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/115143/commits/d84de0792401c5d0a209b8f3f27e12791a0d56ba